### ISSUE TITLE
Tuple addition doesn't result in the correct inferred type

### ISSUE THREAD

**Issue by dabelknap** [astral-sh/ty#4531]
### Summary

When adding (concatenating) tuples it results in an unexpected type. For example `tuple[str, str] + tuple[str, str]` does not resolve to `tuple[str, str, str, str]`.

```python
def myfun(x: tuple[int, int, int, int]):
    print(f"{x=}")

def main() -> None:
    a: tuple[int, int] = (1, 2)
    b: tuple[int, int] = (3, 4)
    c: tuple[int, int, int, int] = a + b # <- results in a ty error

    print(f"{a=}, {b=}") # a=(1, 2), b=(3, 4) 
    print(f"{c=}") # c=(1, 2, 3, 4)

    # results in a ty error
    myfun(a + b) # x=(1, 2, 3, 4)
```

Running `ty check` produces:
```
error[invalid-assignment]: Object of type `tuple[Literal[1, 2, 3, 4], ...]` is not assignable to `tuple[int, int, int, int]`
 --> test.py:7:36
  |
7 |     c: tuple[int, int, int, int] = a + b
  |        -------------------------   ^^^^^ Incompatible value of type `tuple[Literal[1, 2, 3, 4], ...]`
  |        |
  |        Declared type

error[invalid-argument-type]: Argument to function `myfun` is incorrect
  --> test.py:12:11
   |
12 |     myfun(a + b)
   |           ^^^^^ Expected `tuple[int, int, int, int]`, found `tuple[Literal[1, 2, 3, 4], ...]`
info: Function defined here
 --> test.py:1:5
  |
1 | def myfun(x: tuple[int, int, int, int]):
  |     ^^^^^ ---------------------------- Parameter declared here

Found 2 diagnostics
```

### Version

0.0.81

**Comment by carljm** [comment]
Thanks! This is #3413
